src/compiler/program.ts

```diff
@@ -160,15 +162,26 @@ namespace ts {
         host = host || createCompilerHost(options);
 
         let filesByName = createFileMap<SourceFile>(fileName => host.getCanonicalFileName(fileName));
-
-        forEach(rootNames, name => processRootFile(name, /*isDefaultLib:*/ false));
-
-        // Do not process the default library if:
-        //  - The '--noLib' flag is used.
-        //  - A 'no-default-lib' reference comment is encountered in
-        //      processing the root files.
-        if (!skipDefaultLib) {
-            processRootFile(host.getDefaultLibFileName(options), /*isDefaultLib:*/ true);
+        
+        if (oldProgram) {

```

Comment:
Just add an explanatory comment.  Something to give hte intuition for why you are checking these specific options, and the types of options you'd put here in the future.